lib/SILOptimizer/Mandatory/SemanticARCOpts.cpp

```diff
@@ -665,6 +666,84 @@ bool SemanticARCOptVisitor::visitCopyValueInst(CopyValueInst *cvi) {
 //                         load [copy] Optimizations
 //===----------------------------------------------------------------------===//
 
+/// A simple analysis that checks if a specific def (in our case an inout
+/// argument) is ever written to. This is conservative, local and processes
+/// recursively through the def->use.
+static bool defUseIsAddressWrittenTo(SILFunctionArgument *arg) {
+  SmallVector<Operand *, 8> worklist(arg->getUses());
+  while (!worklist.empty()) {
+    auto *op = worklist.pop_back_val();
+    SILInstruction *user = op->getUser();
+
+    if (Projection::isAddressProjection(user)) {
+      for (SILValue r : user->getResults()) {
+        llvm::copy(r->getUses(), std::back_inserter(worklist));
+        continue;
+      }
+    }
+
+    if (auto *oeai = dyn_cast<OpenExistentialAddrInst>(user)) {
+      // Mutable access!
+      if (oeai->getAccessKind() != OpenedExistentialAccess::Immutable) {
+        return true;
+      }
+
+      //  Otherwise, look through it and continue.
+      llvm::copy(oeai->getUses(), std::back_inserter(worklist));
+      continue;
+    }
+
+    // load_borrow and debug_value_addr are both fine as well.
+    if (isa<LoadBorrowInst>(user) || isa<DebugValueAddrInst>(user)) {
```

Comment:
When visiting uses, I like to use `isIncidentalUse`, instead of checking for Debug opcodes so we can add different "marker" instructions later without auditing all passes.

Reply:
Sure.

Reply:
Yes. I was just writing something up real quickly. Also I realized I need to look through access markers.